x/erc20/keeper/msg_server.go

```diff
@@ -411,3 +424,22 @@ func (k Keeper) balanceOf(
 
 	return balance
 }
+
+// monitorApprovalEvent returns an error if the given transactions logs include
+// an unexpected `approve` event
+func (k Keeper) monitorApprovalEvent(msg *evmtypes.MsgEthereumTxResponse) error {
+	LogApprovalSig := []byte("Approval(address,address,uint256)")
+	logApprovalSigHash := crypto.Keccak256Hash(LogApprovalSig)
+	hash := common.BytesToHash([]byte(msg.Hash))
+	logs := k.evmKeeper.GetTxLogsTransient(hash)
+	if len(logs) != 0 {
+		for _, log := range logs {
+			if log.Topics[0].Hex() == logApprovalSigHash.Hex() {
+				return sdkerrors.Wrapf(
+					types.ErrUnexpectedEvent, "unexpected approval event",
+				)
+			}
+		}
+	}
+	return nil
+}
```

Comment:
```suggestion
func (k Keeper) monitorApprovalEvent(res *evmtypes.MsgEthereumTxResponse) error {
	if res == nil || len(res.Logs) == 0 {
		return nil
	}
	logApprovalSig := []byte("Approval(address,address,uint256)")
	logApprovalSigHash := crypto.Keccak256Hash(logApprovalSig)
	hash := common.BytesToHash([]byte(res.Hash))
	for _, log := range res.Logs {
		if log.Topics[0].Hex() == logApprovalSigHash.Hex() {
			return sdkerrors.Wrapf(
				types.ErrUnexpectedEvent, "unexpected approval event",
			)
		}
	}
	return nil
}
```

Reply:
@fedekunze `len(res.Logs) = 0 ` is always true. Seems like the logs are only retrievable from the Transient store.

Reply:
uh ok that's weird, it shouldn't be the case

Reply:
can you add a TODO and create an issue then?

Reply:
https://linear.app/tharsis/issue/ENG-395/callevm-tx-response-logs-is-always-empty